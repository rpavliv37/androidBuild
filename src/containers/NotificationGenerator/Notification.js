import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StatusMessage from '../../components/UIKit/StatusMessage';
/**
 * @module Notification
 */
/**
 * Renders one notification
 * @memberof module:Notification
 */
class Notification extends Component {
  onClick = () => {
    const { deleteNotification, message } = this.props;
    deleteNotification(message.id);
  }

  render() {
    const { message } = this.props;
    const { type, text } = message;

    return (
      <Text type={type} text={text} onClick={this.onClick}>
        {text}
      </Text>
    );
  }
}

Notification.propTypes = {
  message: PropTypes.instanceOf(Object).isRequired,
  deleteNotification: PropTypes.func.isRequired
};

export default Notification;
