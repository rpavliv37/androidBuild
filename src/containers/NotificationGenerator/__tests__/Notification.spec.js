import React from 'react';
import { shallow } from 'enzyme';
import Notification from '../Notification';

describe('NotificationGenerator: Notification', () => {
  it('Shallow render of Notification (success)', () => {
    const props = {
      deleteNotification: () => {},
      message: {
        type: 'success',
        text: 'Test message for notification.'
      }
    };
    const wrapper = shallow(<Notification {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
