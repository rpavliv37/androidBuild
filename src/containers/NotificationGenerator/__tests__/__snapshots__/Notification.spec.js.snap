// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationGenerator: Notification Shallow render of Notification (success) 1`] = `
<StatusMessage
  onClick={[Function]}
  text="Test message for notification."
  type="success"
/>
`;
